.github/workflows/test-check-lint.yml

```diff
@@ -31,6 +31,19 @@ jobs:
 
       - name: Run cpp tests
         run: docker run --rm -e TEST_TMPDIR=/root/.cache_bazel -v $HOME/.cache_bazel_test:/root/.cache_bazel -v "$PWD":/src:ro -w /src --gpus all --privileged spaceandtimelabs/blitzar:12.2.0-cuda-1.71.1-rust-1 /bin/bash -c "cp -av /src/ /src_tmp/; cd /src_tmp; bazel test //..."
+          
+  test-cpp-opt:
+    name: C++ code with optimizations
+    runs-on: self-hosted
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+          # Note: Certain tests crashes nvlink
+          #       See https://developer.nvidia.com/bugs/4288496
+          #       As a simple workaround, I'm disabling for now
+      - name: Run cpp tests with optimization flags on
+        run: docker run --rm -e TEST_TMPDIR=/root/.cache_bazel -v $HOME/.cache_bazel_test:/root/.cache_bazel -v "$PWD":/src:ro -w /src --gpus all --privileged spaceandtimelabs/blitzar:12.2.0-cuda-1.71.1-rust-1 /bin/bash -c "cp -av /src/ /src_tmp/; cd /src_tmp; bazel test -c opt //... -- -//sxt/multiexp/multiproduct_gpu:* -//sxt/algorithm/iteration:*"
```

Comment:
NIT: not a big deal, but maybe we should use `cache_bazel_test_opt` instead of `cache_bazel_test` to avoid re-building files from the previous job.

Reply:
Done. Thank you.